File: 2st/Riyeon/안전지대.py
def solution(board):
    answer = 0
    n = len(board)
    bx = [1, 1, 1, 0, 0, -1, -1, -1]
    by = [1, 0, -1, 1, -1, 1, 0, -1]
    bom = []
    for x in range(n):
        for y in range(n):
            if board[x][y] == 1:
                bom.append((x, y))
    
    for i, j in bom:
        for k in range(8):
            x = i + bx[k]
            y = j + by[k]
            if 0 <= x < n and 0 <= y < n:
                board[x][y] = 1
    
    for i in board:
        answer += i.count(0)
    return answer

File: 4st/Eunseong/안전지대.java
class Solution {
    public int solution(int[][] board) {
        int answer = 0;
        int x[] = {-1, -1, 0, 1, 1, 1, 0, -1};
        int y[] = {0, 1, 1, 1, 0, -1, -1, -1};
        for (int i = 0; i < board.length; i++) {
            for (int j = 0; j < board[i].length; j++) {
                if (board[i][j] == 1) {
                    for (int k = 0; k < x.length; k++) {
                        if (i + y[k] < 0 || j + x[k] < 0 || i + y[k] >= board.length || j + x[k] >= board[i].length) {
                            continue;
                        } else {
                            if (board[i + y[k]][j + x[k]] == 0) {
                                board[i + y[k]][j + x[k]] = 2;
                            }
                        }
                    }
                }
            }
        }
        for (int i = 0; i < board.length; i++) {
            for (int j = 0; j < board[i].length; j++) {
                if (board[i][j] == 0) answer++;
            }
        }
        return answer;
    }
}